packages/wdio-local-runner/src/worker.js

```diff
@@ -63,24 +62,50 @@ export default class WorkerInstance extends EventEmitter {
             runnerEnv.WDIO_LOG_PATH = path.join(this.config.outputDir, `wdio-${cid}.log`)
         }
 
-        log.info(`Start worker ${cid} with arg: ${argv}`)
-        const childProcess = this.childProcess = child.fork(path.join(__dirname, 'run.js'), argv, {
-            cwd: process.cwd(),
-            env: runnerEnv,
-            execArgv,
-            stdio: ['inherit', 'pipe', 'pipe', 'ipc']
-        })
-
-        childProcess.on('message', ::this._handleMessage)
-        childProcess.on('error', ::this._handleError)
-        childProcess.on('exit', ::this._handleExit)
-
-        /* istanbul ignore if */
-        if (!process.env.JEST_WORKER_ID) {
-            childProcess.stdout.pipe(new RunnerTransformStream(cid)).pipe(stdOutStream)
-            childProcess.stderr.pipe(new RunnerTransformStream(cid)).pipe(stdErrStream)
+        // This class emulates a child process. This is not here to stay;
+        // I did this in order to test my assumption that I can get rid of the multiple transpilations
+        // by using the same process. Unfortunately the whole of webdriverio is architected in a way that
+        // it uses inter-process communication via `process.send`. As I wanted to get a changeset for
+        // discussion up first, I decided to minimize the changes and just inject the fake child process
+        // into components that assume to be executed in a child process.
+        class FakeChildProcess {
+            constructor(worker) {
+                this.runner = new Runner(this)
+                this.worker = worker
+                this.runner.on('error', ({ name, message, stack }) => this.worker._handleError({
+                    origin: 'worker',
+                    name: 'error',
+                    content: { name, message, stack }
+                }))
+                this.runner.on('exit', ::this.worker._handleExit)
+            }
+            send(m) {
```

Comment:
Don't use single letter names for arguments, makes it difficult to understand the rest of the code.

Reply:
That code is copied almost verbatim from `run.js`, I didn't touch the car naming - happy to change that though.